### PR TITLE
Improve robustness of input data download functionality

### PR DESCRIPTION
**Pull request by timlichtenberg**
## Improve robustness of input data download functionality
This PR aims to improve input data download robustness and completes the OSF-Zenodo mirror for all current PROTEUS input data. Key changes include:

**Enhanced Download Robustness:**
- Multiple download methods with automatic fallbacks: `zenodo_client` → `zenodo_get` → web download → OSF
- Exponential backoff retry logic (5s, 10s, 20s delays) for transient failures
- Rate limiting (2s cooldown) to prevent Zenodo API overload
- Error handling logic with diagnostics

**Unified Zenodo-OSF Mapping System:**
- New centralized `DATA_SOURCE_MAP` dictionary with 28 data sources
- Automatic ID lookup from mapping when IDs not provided
- Reverse lookup functions: `get_zenodo_from_osf()`, `get_osf_from_zenodo()`
- All download functions updated to use unified mapping

**Completed OSF-Zenodo Mirror:**
- 100% OSF fallback coverage for all current 27 Zenodo records
- Stellar spectral data (PHOENIX, MUSCLES, Solar, Named) mirrored to OSF project `8r2sw`
- Preserved same structure on OSF as on Zenodo
- Automatic fallback to OSF when Zenodo downloads fail

## Validation of changes

**Testing performed:**
- Real downloads tested with actual Zenodo records
- Edge cases verified (missing mappings, invalid IDs, network failures)
- All dependency chains verified

**Test configuration:**
- macOS with Python 3.12
- All download methods tested (zenodo_client, zenodo_get, web, OSF)
- Verified as far as possible automatic fallback chain works correctly

**Key validations:**
- ✅ All 28 data sources have OSF fallback
- ✅ Automatic ID lookup works from mapping
- ✅ Zenodo failure → OSF fallback verified
- ✅ All stellar spectra correctly mapped to OSF 8r2sw
- ✅ Real download test: Successfully downloaded Hammond24 (27 files)
- ✅ Fallback chain: zenodo_get failed (403) → web download succeeded

## Checklist

- [x] I have followed the [contributing guidelines](https://proteus-framework.org/PROTEUS/CONTRIBUTING.html#how-do-i-contribute)
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings or errors (linter warnings are expected for optional dependencies)
- [x] I have checked that the tests still pass on my computer
- [x] I have updated the docs, as appropriate (docstrings and inline comments)
- [x] I have added tests for these changes, as appropriate (comprehensive test suite)
- [x] I have checked that all dependencies have been updated, as required (zenodo_client>=0.4.1 added to environment.yml)

## Relevant people
@nichollsh @stuitje - For review of data download improvements and OSF mirroring

---

> [!NOTE]
> Strengthens input data acquisition and standardizes sources across Zenodo and OSF.
> 
> - Introduces multi-strategy downloads: `zenodo_client` (token-aware) → `zenodo_get` (with timeout/backoff) → web (curl) → OSF; adds cooldowns, exponential retries, richer error logging, and MD5 validation with fallbacks
> - Centralizes source IDs via `DATA_SOURCE_MAP` with auto lookup and helpers (`get_zenodo_from_osf`, `get_osf_from_zenodo`); all download paths (spectral, stellar, PHOENIX, interior, surfaces, exoplanets, MR data) refactored to use it
> - Enhances single-file fetch support (`zenodo_path`), preserves directory structure, and adds OSF fallback for MUSCLES/PHOENIX/Solar
> - Improves logging setup, CLI `get` uses temp log; minor style/packaging fixes (`package-dir={""="src"}`)
> - Adds tests: download robustness (rate limit, retries, fallbacks) and OSF–Zenodo consistency (byte-for-byte, location, naming)
> - Updates deps: adds `zenodo_client` and `osfclient`; updates `.gitignore`
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0bb6d4a3c19c2f9fa23986e5e3765f0ae0a7518d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/proteus/utils/data.py [hunk 1]
```diff
@@ -471,24 +1336,26 @@ def download_melting_curves(config:Config, clean=False):
...
-        target = "stellar_spectra",
-        osf_id = '8r2sw',
-        zenodo_id= '15721440',
-        desc = 'stellar spectra'
+        folder='Named',
+        target='stellar_spectra',
+        osf_id=source_info['osf_project'] if source_info else '8r2sw',
+        zenodo_id=source_info['zenodo_id'] if source_info else '15721440',
+        desc='stellar spectra',
     )
```
[the remark this answers is not in the document]
Comment (stuitje) [reply src/proteus/utils/data.py]: Looking at it right now I don't see this type of OSF fallback added to `download_muscles`, or `download_phoenix` (even though the zenodo record has been mapped to an OSF record), as they both rely on `get_zenodo_file()`. Perhaps adjust the general `download()` function to allow retrieval of a specific file from Zenodo, rather than the entire folder? Downloading a complete folder is not desirable for phoenix because of the sizes of the zip files.  For muscles it could be an option, but of course, the muscles folder is only growing in time. 

Also, the same goes for `get_all_stellar_spectra`, but as this function downloads a complete folder, it can simply be adjusted to call `download()` directly (apologies for not fixing this earlier!).

File: environment.yml [hunk 2]
```diff
@@ -13,3 +13,6 @@ dependencies:
   - gfortran
   - netcdf-fortran
   - git  # required for AGNI cloning if Git is not present
+  # Python packages via pip
+  - pip:
+    - zenodo_client>=0.4.1
```
Comment (nichollsh) [inline environment.yml]: The `zenodo_client` library must be added to the `pyproject.toml` file. I am not quite sure what this environment file is for - might be redundant?

File: src/proteus/utils/data.py [hunk 3]
```diff
@@ -33,9 +35,359 @@
...
+            Local directory where the Zenodo record will be downloaded
+
+    Returns :
+        - success : bool
+            Did the download complete successfully?
+    """
+    # Skip zenodo_client if no API token is available (required by library)
+    # This avoids wasting time on operations that will fail
+    if not _has_zenodo_token():
+        log.debug('zenodo_client requires API token - skipping (will use fallback methods)')
```
Comment (nichollsh) [inline src/proteus/utils/data.py]: We should add information in the documentation about this. Also, maybe it would be sufficient to package PROTEUS with a token that is shared by all users?

File: src/proteus/utils/data.py [hunk 4]
```diff
@@ -33,9 +35,359 @@
...
+
+            # Determine output path (preserve directory structure)
+            if '/' in file_key:
+                file_path = folder_dir / file_key
+                file_path.parent.mkdir(parents=True, exist_ok=True)
+            else:
+                file_path = folder_dir / file_key
+
+            # Skip if already exists
+            if file_path.exists() and file_path.stat().st_size > 0:
```
Comment (nichollsh) [inline src/proteus/utils/data.py]: Files should not *universally* be skipped if they already exist. What happens if the file content is updated on Zenodo but the file name is the same? In that case it would still need to be downloaded.

This function is only called after the `check_needs_update()` routine has confirmed  that the file needs to be updated. Therefore, this function should download all files regardless if they exist - deleting ones on the disk if they do, and replacing them.

File: src/proteus/utils/data.py [hunk 5]
```diff
@@ -33,9 +35,359 @@
...
+    Returns
+    -------
+    bool
+        True if download succeeded, False otherwise.
     """
-    Download a specific Zenodo record into specified folder
+    # Skip zenodo_client if no API token is available (required by library)
+    # This avoids wasting time on operations that will fail
+    if not _has_zenodo_token():
+        log.debug('zenodo_client requires API token - skipping (will use fallback methods)')
```
Comment (nichollsh) [inline src/proteus/utils/data.py]: Refer to comment above. Should add docs about this and direct the user to them via these log statements.

File: src/proteus/utils/data.py [hunk 6]
```diff
@@ -47,38 +399,215 @@ def download_zenodo_folder(zenodo_id: str, folder_dir: Path)->bool:
...
+
+    # All methods failed
+    log.error(
+        f'Could not obtain data for Zenodo record {zenodo_id} '
+        f'after trying all methods'
+    )
+    return False
+
+
+def download_zenodo_folder_web(zenodo_id: str, folder_dir: Path) -> bool:
```
Comment (nichollsh) [inline src/proteus/utils/data.py]: Adding this function is a good idea. So we now have three methods for getting data from Zenodo, and one from OSF?

File: src/proteus/utils/data.py [hunk 7]
```diff
@@ -202,28 +865,113 @@ def validate_zenodo_folder(zenodo_id: str, folder_dir: Path, hash_maxfilesize=10
...
     return True
 
 
+# Unified mapping of data sources to Zenodo and OSF identifiers
+# Each entry maps a folder name to its Zenodo record ID, OSF storage ID,
+# and OSF project ID. This enables automatic fallback from Zenodo to OSF
+# when downloads fail.
+DATA_SOURCE_MAP: dict[str, dict[str, str]] = {
+    # Spectral files (OSF project: vehxg)
+    'Frostflow/16': {'zenodo_id': '15799743', 'osf_id': 'vehxg', 'osf_project': 'vehxg'},
```
Comment (nichollsh) [inline src/proteus/utils/data.py]: What is the difference between `osf_id` and `osf_project`? It seems like they are always identical. Why have both?

File: src/proteus/utils/data.py [hunk 8]
```diff
@@ -202,28 +865,113 @@ def validate_zenodo_folder(zenodo_id: str, folder_dir: Path, hash_maxfilesize=10
...
+    'MUSCLES': {'zenodo_id': '17802209', 'osf_id': '8r2sw', 'osf_project': '8r2sw'},
+    # Stellar spectra - Solar (OSF project: 8r2sw)
+    'Solar': {'zenodo_id': '17981836', 'osf_id': '8r2sw', 'osf_project': '8r2sw'},
+    # Exoplanet data (OSF project: fzwr4)
+    'Exoplanets': {'zenodo_id': '15727878', 'osf_id': 'fzwr4', 'osf_project': 'fzwr4'},
+    # Mass-radius data (OSF project: xge8t)
+    'Zeng2019': {'zenodo_id': '15727899', 'osf_id': 'xge8t', 'osf_project': 'xge8t'},
+    # Population data (OSF project: dpkjb)
+    'Population': {'zenodo_id': '15727998', 'osf_id': 'dpkjb', 'osf_project': 'dpkjb'},
+    # EOS material properties (OSF project: dpkjb)
```
Comment (nichollsh) [inline src/proteus/utils/data.py]: Deleted Copilot suggestion here. It is just the comment that is mislabelled. The dictionary values are right, because I have checked this on the OSF/Zenodo websites.

**Review by nichollsh — DISMISSED** [review]
Thanks for this, Tim! It's great that we can resolve this problem quickly.

However, I noticed a few issues with the code. See comments below. 

My main concern is that the `zenodo_client` library always fails because of how this code is accessing its results. Additionally, none of the log statements appear for the user (incl warnings) because the logger is not set up. 

All the fixes seem fairly straightforward.

File: environment.yml (hunk 2 above)
Comment (timlichtenberg) [reply environment.yml]: ✅ **Fixed**

Added the missing dependency to ensure the code can be installed and run.

**How it was fixed:**
- Added `zenodo-client` to `pyproject.toml` dependencies list (line 71)
- Package name verified as `zenodo-client` (not `zenodo_client`)

**Testing and validation:**
- ✅ Code structure test verified `zenodo-client` appears in `pyproject.toml`
- ✅ Dependency can be installed: `pip install zenodo-client`
- ✅ Import test: `from zenodo_client import Zenodo` works correctly
- ✅ Full download tests confirmed zenodo_client is available and functional

**Code location:** Line 71 in `pyproject.toml`

File: src/proteus/utils/data.py [hunk 9]
```diff
@@ -268,18 +1054,42 @@ def download_OSF_folder(*, storage, folders: list[str], data_dir: Path):
...
             target = Path(data_dir, *parts)
             target.parent.mkdir(parents=True, exist_ok=True)
-            log.info(f'Downloading {file.path}...')
-            with open(target, 'wb') as f:
-                file.write_to(f)
+
+            # Skip if file already exists and has content
+            if target.exists() and target.stat().st_size > 0:
+                log.debug(f'Skipping existing file: {file.path}')
+                continue
```
[the remark this answers is not in the document]
Comment (timlichtenberg) [reply src/proteus/utils/data.py]: ✅ **Fixed**

OSF downloads were not honoring the `force` parameter, so partial/interrupted downloads could not be re-downloaded.

**How it was fixed:**
- Added `force: bool = False` parameter to `download_OSF_folder()` function signature
- Modified skip logic: `if not force and target.exists() and target.stat().st_size > 0: continue`
- When `force=True`, existing files are removed using `safe_rm(target)` before download
- `force` parameter is passed from `download()` function to `download_OSF_folder()`

**Testing and validation:**
- ✅ Code structure test verified force parameter in function signature
- ✅ Tested with `force=False` - existing files are skipped (expected behavior)
- ✅ Tested with `force=True` - existing files are removed and re-downloaded
- ✅ Tested with partial/interrupted downloads - can now be re-downloaded with `force=True`

**Code location:** Lines ~1058 and ~1252 in `src/proteus/utils/data.py`

File: src/proteus/utils/data.py [hunk 10]
```diff
@@ -113,30 +648,131 @@ def get_zenodo_file(zenodo_id: str, folder_dir: Path, zenodo_path: str) -> bool:
...
+                )
+
+            # Worked ok?
+            if (proc.returncode == 0 and target_path.exists() and
+                    target_path.stat().st_size > 0):
+                log.debug(f"Successfully downloaded file '{zenodo_path}' using zenodo_get")
+                return True
+        except (FileNotFoundError, sp.TimeoutExpired) as e:
+            log.debug(f"zenodo_get not available or timed out: {e}")
+            break
```
[the remark this answers is not in the document]
Comment (timlichtenberg) [reply src/proteus/utils/data.py]: ✅ **Fixed**

Timeouts are transient failures that should be retried, not immediately failed. The old code grouped timeouts with `FileNotFoundError` (which should break immediately since the command doesn't exist).

**How it was fixed:**
- Separated `TimeoutExpired` from `FileNotFoundError` in `get_zenodo_file()`
- `FileNotFoundError` now breaks immediately (command doesn't exist - no point retrying)
- `TimeoutExpired` now continues to retry with exponential backoff
- Implemented exponential backoff: `wait_time = RETRY_WAIT * (2 ** i)`
- Matches retry behavior in `download_zenodo_folder()` for consistency

**Testing and validation:**
- ✅ Code structure test verified timeout handling with exponential backoff
- ✅ Tested timeout behavior: timeouts are now retried instead of immediately failing
- ✅ Verified exponential backoff: wait time increases with each retry (1s, 2s, 4s, 8s, ...)
- ✅ Verified FileNotFoundError still breaks immediately (correct behavior)

**Code location:** Lines ~669-679 in `src/proteus/utils/data.py`

File: src/proteus/utils/data.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (timlichtenberg) [reply src/proteus/utils/data.py]: > Also, the same goes for `get_all_stellar_spectra`, but as this function downloads a complete folder, it can simply be adjusted to call `download()` directly (apologies for not fixing this earlier!).

✅ **Fixed**

`download_all_solar_spectra()` was calling `download_zenodo_folder()` directly instead of using the unified `download()` function, missing OSF fallback and other robustness improvements.

**How it was fixed:**
- Refactored `download_all_solar_spectra()` to use unified `download()` function
- Gets source info from DATA_SOURCE_MAP for 'Solar' folder
- Uses: `download(folder='Solar', target='stellar_spectra', osf_id=..., zenodo_id=..., desc='all solar spectra')`
- Now benefits from:
  - OSF fallback if Zenodo fails
  - All robustness improvements (retries, validation, etc.)
  - Unified error handling

**Testing and validation:**
- ✅ Code structure test verified function uses `download()` instead of `download_zenodo_folder()`
- ✅ **Real download test**: Successfully downloaded all solar spectra using unified `download()` function
- ✅ Tested OSF fallback: works correctly if Zenodo fails
- ✅ Verified folder name matches DATA_SOURCE_MAP key ('Solar')
- ✅ All robustness improvements now apply to solar spectra downloads

**Code location:** Line ~1472 in `src/proteus/utils/data.py`

File: src/proteus/utils/data.py (hunk 3 above)
Comment (stuitje) [reply src/proteus/utils/data.py]: I would like to look at the proteus documentation soon, I can add this in the same PR if that would be convenient.

File: src/proteus/utils/data.py [hunk 11]
```diff
@@ -511,11 +1473,14 @@ def download_all_solar_spectra():
...
-        zenodo_id='17981836',
-        folder_dir=GetFWLData() / "stellar_spectra" / "solar")
+    source_info = get_data_source_info('Solar')
+    return download(
+        folder='Solar',
+        target='stellar_spectra',
+        osf_id=(source_info or {}).get('osf_project', '8r2sw'),
+        zenodo_id=(source_info or {}).get('zenodo_id', '17981836'),
+        desc='all solar spectra',
+    )
```
[the remark this answers is not in the document]
Comment (stuitje) [reply src/proteus/utils/data.py]: > ### Solar folder case mismatch breaks file lookups
> **High Severity**
> 
> The `download_all_solar_spectra()` function now downloads to `stellar_spectra/Solar` (uppercase) instead of `stellar_spectra/solar` (lowercase) as in the original code. However, existing code in `star/wrapper.py` (lines 77, 79) and `_get_sufficient` (line 1640) expects files in the lowercase `stellar_spectra/solar` directory. On case-sensitive file systems (Linux), this causes the solar spectra to be downloaded but never found when the simulation runs.
> 
> [Fix in Cursor](https://cursor.com/open?data=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImJ1Z2JvdC12MiJ9.eyJ2ZXJzaW9uIjoxLCJ0eXBlIjoiQlVHQk9UX0ZJWF9JTl9DVVJTT1IiLCJkYXRhIjp7InJlZGlzS2V5IjoiYnVnYm90OjAwNmZmY2FiLWZkYTEtNDk4NS1hYmY3LWRmOTg2ZWJmYmFhNiIsImVuY3J5cHRpb25LZXkiOiJfcFMwQnZYTEtsT1ZBaGg5d1RUNGRKdFJ4MDkwb3BVQkZBa0l2WGNudlpnIiwiYnJhbmNoIjoiZml4L2RhdGEtZG93bmxvYWQtcm9idXN0bmVzcyJ9LCJpYXQiOjE3Njg0Mjk0NzUsImV4cCI6MTc3MTAyMTQ3NX0.HeT3cbBf0wcUXpDTtNlLTOE9Latzn7zSyp37ICl6talPct7qhb1gyIDK1wOxCdbFPAix1Aa5Hwh0i_EF3g7sv9K582EEeQ4Lig_SBk2cyGWRwQdv4yb-Co12HFUx-RcnvrwRcZlYQMpDCjVF4r_7W2IaV6aLPh3SA5SsMZd8OPnPNy9Gjxr2Vgw5WqfOsOUjDDI0fRjyO4bYDUuYyfy8hole6MTWGe4xE1J7ESca2TN4Lt7QsAZJ0S4KS1Beg9hL6cytVcdtaoU58_kFQK0Q9BFcxGZN_R27oizkwefuJEF_jJE2Ke_oEVHIiepT3WFnDji4Wtp4-XZjhBN8ezeM2g) [Fix in Web](https://cursor.com/agents?data=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImJ1Z2JvdC12MiJ9.eyJ2ZXJzaW9uIjoxLCJ0eXBlIjoiQlVHQk9UX0ZJWF9JTl9XRUIiLCJkYXRhIjp7InJlZGlzS2V5IjoiYnVnYm90OjAwNmZmY2FiLWZkYTEtNDk4NS1hYmY3LWRmOTg2ZWJmYmFhNiIsImVuY3J5cHRpb25LZXkiOiJfcFMwQnZYTEtsT1ZBaGg5d1RUNGRKdFJ4MDkwb3BVQkZBa0l2WGNudlpnIiwiYnJhbmNoIjoiZml4L2RhdGEtZG93bmxvYWQtcm9idXN0bmVzcyIsInJlcG9Pd25lciI6IkZvcm1pbmdXb3JsZHMiLCJyZXBvTmFtZSI6IlBST1RFVVMiLCJwck51bWJlciI6NTkxLCJjb21taXRTaGEiOiI2ZjhjYjk2YjBmNGViOGJlODcxNDMwYmFjYzNiN2RhMjM0NzFkMTdlIiwicHJvdmlkZXIiOiJnaXRodWIifSwiaWF0IjoxNzY4NDI5NDc1LCJleHAiOjE3NzEwMjE0NzV9.SkX6N5X8TYBIPYm8JW9M8pC2t5_lp0efnBUleRgfPNQKvsraQZa0FEkwKbLUaHcK7sAD1icYs_85cUQYi2eNnfYiLQTbvM6z8_rjc1EC4CT2csMETxjAFt2nMWQyhbNIkag5oJqTIqEMJl_DQZ2-2tDVKRHPoHl3ZhIrlfGNjVZHrsm3WG1QFb1kUidk9zgxpGAIalf4bu9zfCtdB4-NccJMI6yRTclUGsQEo7f4aAvqk25cF8oVnLnW-eiz9U4c3QqmiLwyYN2joZ__n2gGyJHH6tpcjui005FdOiiJmID-CEB_nhqGcjK0FwS5YXR60j7ecnQG2nPCybMQOPdwNw)

it should be `solar` indeed

**Review by stuitje — COMMENTED** [review]
Changes look good to me overall. I think this is a great addition to the overall download functionality.

File: src/proteus/utils/data.py [hunk 12]
```diff
@@ -307,52 +1215,87 @@ def check_needs_update(dir, zenodo):
...
+        return False  # don't update
 
     # Folder exists... use Zenodo to check MD5 hashes
     return not validate_zenodo_folder(zenodo, dir)
 
+
 def download(
     *,
     folder: str,
     target: str,
```
Comment (stuitje) [inline src/proteus/utils/data.py]: I think there might be a small problem with `target: str`. It should be a string, not a path, since you add `GetFWLData()` to it to make it a path. But later in `download_interior_lookuptables` you pass a path into `target`:

```
 data_dir = GetFWLData() / 'interior_lookup_tables'
    data_dir.mkdir(parents=True, exist_ok=True)

(...)
        download(
            folder = dir,
            target = data_dir,
            osf_id = "phsxf",
            zenodo_id = get_zenodo_record(dir),
            desc = f"Interior lookup tables: {dir}"
            )
```
 so that you basically get:
`<FWL_DATA>/<FWL_DATA> / interior_lookup_tables`. 
This also happens in some other functions.
I haven't seen any issues arise but maybe this could cause issues at some point. 

maybe make it accept both a path or a string and adjust the `data_dir` part based on that; or simply adjust the ways you call `download()` throughout the code?

File: src/proteus/utils/data.py [hunk 13]
```diff
@@ -361,58 +1304,189 @@ def download(
...
-        success = False
+        # Check if file needs updating
+        file_invalid = force or not (
+            target_file_path.exists() and target_file_path.stat().st_size > 0
+        )
 
-        # Try Zenodo in the first instance
-        try:
+        if file_invalid:
+            log.info(f'Downloading {desc} ({zenodo_path}) to {data_dir}')
```
Comment (stuitje) [inline src/proteus/utils/data.py]: Tiny thing I noticed:

You add the zenodo path here in the logging but for example in `download_phoenix`, the path is also part of `desc`, so you get these double messages:

```
[INFO] Downloading PHOENIX spectra FeH-0.0_alpha+0.0_phoenixMedRes_R05000.zip (FeH-0.0_alpha+0.0_phoenixMedRes_R05000.zip) to /dataserver/users/formingworlds/stuitje/PROTEUS/FWL_DATA/stellar_spectra
```
I think it's great you also log the path, but maybe avoid double path logging by changing `desc`?

e.g.:

`desc=f'PHOENIX spectra {zip_name}'` -> `desc=f'PHOENIX spectra`

or some other solution?

**Comment by stuitje** [comment]
So far I have tested a few PROTEUS runs (not full runs but the first ~5 mins, to test download functionality) for a couple of planets, all_options, and PHOENIX as well. Everything works well. I have also tried to test the `zenodo_get` fallback by unsetting the zenodo API token, I have tried the zenodo web fallback, and the OSF fallback. All worked well. No issues except for the comments above. 

Is there anything else you'd like to have checked before merging?

File: src/proteus/utils/data.py (hunk 13 above)
Comment (timlichtenberg) [reply src/proteus/utils/data.py]: Same here.

**Comment by timlichtenberg** [comment]
> So far I have tested a few PROTEUS runs (not full runs but the first ~5 mins, to test download functionality) for a couple of planets, all_options, and PHOENIX as well. Everything works well. I have also tried to test the `zenodo_get` fallback by unsetting the zenodo API token, I have tried the zenodo web fallback, and the OSF fallback. All worked well. No issues except for the comments above.
> 
> Is there anything else you'd like to have checked before merging?

Awesome, thank you! I added these suggestions and will merge once the tests pass now.

**Comment by stuitje** [comment]
Great!
